### PR TITLE
Feat/validate shlink

### PR DESCRIPTION
**Pull request by jacob-khoza-symb**
## Summary by CodeRabbit

- **New Features**
	- Introduced validation for shared links, enhancing security by requiring a management token and passcode.
	- Added optional passcode property to the shared link request structure for added flexibility.
	- Implemented a new error handling mechanism for specific validation errors related to shared links.

- **Bug Fixes**
	- Improved error reporting for validation issues, providing clearer feedback for specific errors.

- **Tests**
	- Added comprehensive unit tests for the new shared link validation logic, covering various scenarios to ensure reliability.

### CODE REVIEW

File: src/usecases/shlinks/validate-shlink.test.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        new Date(Date.now() - 10000), // past date
+        'link-id'
+      );
+      validateSHLinkUseCase({ shlink: invalidSHLink, passcode: 'wrong-passcode' });
+    } catch (error) {
+      expect(error).toBeInstanceOf(SHLinkValidationError);
+      expect(error.message).toBe('Inactive share link access');
+      expect(error.code).toBe(403);
+    }
+  });
```
Comment (BMartinos) [inline src/usecases/shlinks/validate-shlink.test.ts:92]: Wondering if we should also maybe add a test to determine a link is valid without an expiry date? Is the expiry date defined as blank or as an unreachable future date when not set?

File: src/usecases/shlinks/validate-shlink.test.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      'unique-user-id',
+      5,
+      true,
+      'management-token',
+      'valid-passcode',
+      new Date(Date.now() - 10000), // past date
+      'link-id'
+    );
+    expect(() => validateSHLinkUseCase({ shlink: expiredSHLink, passcode: 'valid-passcode' }))
+      .toThrow(SHLinkValidationError);
```
Comment (BMartinos) [inline src/usecases/shlinks/validate-shlink.test.ts:44]: Question/Suggestion. Im wondering if it makes sense to cater for a few different statuses returned within the validation error. The 403 is applicable for most of the validation failures, but wondering around the case where the link is expired, or disabled, that the code should maybe be something like `410 Gone` as this link still exists, but will never be accessible again for retrieval

**Review by BMartinos — APPROVED** [review]
@jacob-khoza-symb overall looks great, Just left a comment/suggestion but if its not applicable, its not blocking the PR from being merged

File: src/usecases/shlinks/validate-shlink.test.ts (hunk 1 above)
Comment (jacob-khoza-symb) [reply src/usecases/shlinks/validate-shlink.test.ts:106]: Added the test.

The expiry date is defined as blank when not set.

File: src/usecases/shlinks/validate-shlink.test.ts (hunk 2 above)
Comment (jacob-khoza-symb) [reply src/usecases/shlinks/validate-shlink.test.ts:44]: Added 410 for situations where the share link is disabled or expired.

**Review by BMartinos — APPROVED** [review]
LGTM :+1:

**Pull request merged**